### ISSUE TITLE
HS13M - probably works the same

### ISSUE THREAD

**Issue by SaiCode-DEV** [iMicknl/LoctekMotion_IoT#66]
I bought the Flexispot E7 Pro and it has the HS12M remothe. is there a simple way to test/check if the protocol changed?

**Comment by iMicknl** [comment]
Not really. But if you have a RJ45 port on your controller, big chance protocol is supported. You could also open your display/remote and see what the pinout is.

**Comment by SaiCode-DEV** [comment]
![HS13M](https://github.com/iMicknl/LoctekMotion_IoT/assets/53049957/9badef71-8427-4afd-997d-31b70c4346c0)

Sadly no spesific Information under the cables

**Comment by iMicknl** [comment]
@SaiCode-DEV the easiest way to test this is to just hook up an ESP32 to your control box. See https://github.com/iMicknl/LoctekMotion_IoT/issues/67.

**Comment by MauiR** [comment]
Hello, I also own the Flexispot E7 Pro desk. Unfortunately, with the new Office-desk.yaml configuration in ESPHome (https://github.com/iMicknl/LoctekMotion_IoT/blob/6a1da5dc22688cd4eb8ca8e90e5c4b123c4354d9/packages/office-desk.yaml), only the desk height functionality is working. All other features such as Alarm, Desk position, Preset 1, Preset 2, Sit, and Stand are non-functional.
The "desk height" however, is working perfectly and updates in real-time

I've utilized an ESP32 with this RJ45 connector layout: https://github.com/iMicknl/LoctekMotion_IoT?tab=readme-ov-file#hs01b-1


![image](https://github.com/iMicknl/LoctekMotion_IoT/assets/86602061/7669a305-a69c-46ed-95c0-4e3b473436c1)

**Comment by iMicknl** [comment]
@MauiR have you tried the new configuration and pin-out from https://github.com/iMicknl/LoctekMotion_IoT/issues/67?

**Comment by MauiR** [comment]
> @MauiR have you tried the new configuration and pin-out from #67?

Yes, I've chosen the new configuration pinout. However, I haven't connected SWIM and RES because there were no specifications provided for them on the ESP32.

**Comment by MauiR** [comment]
I contacted FlexiSpot to inquire if they have a new protocol and if they could provide information about it. Unfortunately, they weren't very cooperative and were unwilling to assist.

**Comment by MauiR** [comment]
I can say that I have now managed to get it running with the Flexispot 7 Pro. However, it only works with an ESP8266. I haven't been able to get it running with an ESP32. I'll take another closer look.

**Comment by Rudd-O** [comment]
I'm interested in this too.  Pinout and software.  How can I help?

**Comment by locnh** [comment]
I have an `E7Q` with `HS13M` keypad, work smoothly with these boards:
- `esp32s3-devkitc-1`
- `esp32c6-devkitc-1`
- `esp32-mini-d1`

**Comment by dimitri-vs** [comment]
Can confirm HS13M-1C0 works. Running it on an E7 Pro Plus with CB38M2L and a XIAO ESP32C6, connected to the spare RJ45 port (physical keypad stays on the primary port, both work simultaneously).

All commands functional: Stand, Sit, Preset 1, Preset 2, Up, Down, etc. Height sensor reads the 7-segment display correctly. Same `9B...9D` packet format at 9600 baud.

One gotcha on HS13M specifically - if you're getting height data but commands are ignored, it's a state/timing issue with newer controllers, not a protocol difference. These need a proper PIN 20 wake sequence (1s hold, not a quick pulse), and commands need to go out as responses to the controller's `0x11` polls rather than fire-and-forget. Once I got that right, everything worked first try.
